app/actions/websocket/index.ts

```diff
@@ -64,90 +68,119 @@ export async function handleClose(serverUrl: string, lastDisconnect: number) {
-                    // eslint-disable-next-line no-console
-                    console.log('FAILED TO BATCH CHANNELS');
+    // if no longer a member of the current team
+    if (initialTeamId !== system.currentTeamId) {
+        let cId = await queryCurrentChannelId(database);
+        if (tabletDevice) {
+            if (!cId) {
+                const channel = await queryDefaultChannelForTeam(database, initialTeamId);
+                if (channel) {
+                    cId = channel.id;
                 }
             }
+
+            switchToChannelById(serverUrl, cId, initialTeamId);
         }
+    }
+
+    let removeTeams;
+    if (removeTeamIds?.length) {
+        // Immediately delete myTeams so that the UI renders only teams the user is a member of.
+        removeTeams = await queryTeamsById(database, removeTeamIds);
+        await deleteMyTeams(operator, removeTeams!);
+    }
+
+    let removeChannels;
+    if (removeChannelIds?.length) {
+        removeChannels = await queryChannelsById(database, removeChannelIds);
+    }
+
+    const modelPromises = await prepareModels({operator, initialTeamId, removeTeams, removeChannels, teamData, chData, prefData, meData});
+    if (rolesData.roles?.length) {
+        modelPromises.push(operator.handleRole({roles: rolesData.roles, prepareRecordsOnly: true}));
+    }
 
-        channelMemberships = memberships;
-
-        if (currentChannelId) {
-            const stillMemberOfCurrentChannel = memberships?.find((cm) => cm.channel_id === currentChannelId);
-            const channelStillExist = channels?.find((c) => c.id === currentChannelId);
-            const viewArchivedChannels = config?.ExperimentalViewArchivedChannels === 'true';
-
-            if (!stillMemberOfCurrentChannel) {
-                handleUserRemovedFromChannelEvent(serverUrl, {data: {user_id: currentUserId, channel_id: currentChannelId}});
-            } else if (!channelStillExist ||
-                (!viewArchivedChannels && channelStillExist.delete_at !== 0)
-            ) {
-                handleChannelDeletedEvent(serverUrl, {data: {user_id: currentUserId, channel_id: currentChannelId}} as WebSocketMessage);
-            } else {
-                // TODO Differentiate between post and thread, to fetch the thread posts
-                fetchPostsSince(serverUrl, currentChannelId, lastDisconnectedAt);
+    if (modelPromises.length) {
+        const models = await Promise.all(modelPromises);
+        const flattenedModels = models.flat();
+        if (flattenedModels?.length > 0) {
+            try {
+                await operator.batchRecords(flattenedModels);
+            } catch {
+                // eslint-disable-next-line no-console
+                console.log('FAILED TO BATCH WS reconnection');
             }
         }
+    }
+
+    const currentChannelId = await queryCurrentChannelId(database);
+    if (currentChannelId) {
+        // TODO Differentiate between post and thread, to fetch the thread posts
```

Comment:
Ticket number for all todos

Reply:
Tickets do exist, I just moved the lines , but sure I'll add the ticket numbers